server/tests/factory/user-factory.js

```diff
@@ -11,14 +11,14 @@ const generateToken = async (userDetails) => {
 };
 
 const createTestUser = async ({
-  username, email, password
+  username, email, password, active
 }) => {
   const newUser = await User.create({
     id: faker.random.uuid(),
     username: username || faker.random.alphaNumeric(6),
     email: email || faker.internet.email(),
     password: password || faker.internet.password(),
-    active: true
+    active: active || false
```

Comment:
No please,

create an extra factory function for non active users and use that for your tests.